scripts/generate_lv_conf.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import argparse
+
+DIR_SCRIPTS = os.path.dirname(__file__)
+REPO_ROOT = os.path.join(DIR_SCRIPTS, "..")
+DIR_CWD = os.getcwd()
+
+
+def fatal(msg):
+    print("ERROR! " + msg)
+    exit(1)
+
+
+def get_args():
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter, description=""
+                                     "Update the configuration file of your existing project based on default parameters and latest LVGL template. Eg.:\n"
+                                     " python3 generate_lv_conf.py ./src \n"
+                                     " python3 generate_lv_conf.py --template modules/lvgl/lv_conf_template.h ./my_config_folder"
+                                     )
+
+    parser.add_argument('--template', type=str, default=REPO_ROOT, nargs='?',
+                        help='Path of "lv_conf_template.h" file or the folder containing it\n(optional, 1 folder above the python script by default)')
+
+    parser.add_argument('--config', type=str, default=None, nargs='?',
+                        help='Path of "lv_conf.h" file (optional)')
+
+    parser.add_argument('--defaults', type=str, default=None, nargs='?',
+                        help='Path of "lv_conf.defaults" file (optional)')
+
+    parser.add_argument('target', metavar='target', type=str, default=DIR_CWD, nargs='?',
+                        help='Folder containing "lv_conf.h" and "lv_conf.defaults" files\n(optional, current work folder by default)')
+
+    args = parser.parse_args()
+
+    if os.path.isdir(args.template):
+        args.template = os.path.join(args.template, "lv_conf_template.h")
+
+    if not args.config:
+        args.config = os.path.join(args.target, "lv_conf.h")
+
+    if not args.defaults:
+        args.defaults = os.path.join(args.target, "lv_conf.defaults")
+
+    if not os.path.exists(args.template):
+        fatal(f"Template file not found at {args.template}")
+    if not os.path.exists(args.config):
+        fatal(f"User config file not found at {args.config}")
+    if not os.path.exists(args.defaults):
+        fatal(f"User defaults not found at {args.default}")
+
+    return args
+
+
+def parse_defaults(path: str):
+    defaults = {}
+
+    with open(path, 'r', encoding='utf-8') as file:
+        for line in file.readlines():
+            groups = re.search(r'([A-Z0-9_]+)\s+(.+)', line).groups()
```

Comment:
May need to check basic error like empty lines.

Reply:
Good catch